File: app/src/main/java/org/tasks/injection/InjectingAppWidgetProvider.java
package org.tasks.injection;

import android.appwidget.AppWidgetProvider;
import android.content.Context;
import android.content.Intent;

public abstract class InjectingAppWidgetProvider extends AppWidgetProvider {

  @Override
  public void onReceive(Context context, Intent intent) {
    inject(
        ((InjectingApplication) context.getApplicationContext())
            .getComponent()
            .plus(new BroadcastModule()));

    super.onReceive(context, intent);
  }

  protected abstract void inject(BroadcastComponent component);
}

File: app/src/main/java/org/tasks/injection/InjectingAppWidgetProvider.kt
package org.tasks.injection

import android.appwidget.AppWidgetProvider
import android.content.Context
import android.content.Intent

abstract class InjectingAppWidgetProvider : AppWidgetProvider() {
    override fun onReceive(context: Context, intent: Intent) {
        inject(
                (context.applicationContext as InjectingApplication)
                        .component
                        .plus(BroadcastModule()))
        super.onReceive(context, intent)
    }

    protected abstract fun inject(component: BroadcastComponent)
}